app/models/vendor.rb

```diff
@@ -0,0 +1,47 @@
+class Vendor < ActiveRecord::Base
+  belongs_to :market
+  has_many :sales
+  has_many :products
+
+  validates_presence_of :name, :number_of_employees, :market_id
+  validates_numericality_of :number_of_employees, :market_id
+
+  def recent_sales_first(how_many = sales.count)
+    sorted_sales = sales.sort_by { |sale| sale.purchase_time }
+    sorted_sales.reverse!
+
+    return sorted_sales[0, how_many]
+  end
+
+  def total_sales
+    amounts = sales.map { |sale| sale.amount }
+    total = amounts.reduce { |sum, n| sum + n }
+
+    return total
+  end
+
+  def month_sales
+    current_month = Time.now.month
+    current_month_sales = []
+
+    sales.each do |sale|
+      current_month_sales.push(sale) if sale.purchase_time.month == current_month
+    end
+
+    return current_month_sales
```

Comment:
Try out [`select`](http://ruby-doc.org/core-2.2.2/Enumerable.html#method-i-select) instead of `each` here. Then you can build your array in one go without creating an outside one that you're pushing into.

Also, you might also want to try leveraging an ActiveRecord method like [`where`](http://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-where) for something like this in the future.